strawberry/extensions/tracing/datadog.py

```diff
@@ -0,0 +1,80 @@
+from inspect import isawaitable
+from typing import Optional
+
+from ddtrace import tracer
+
+from strawberry.extensions import Extension
+from strawberry.extensions.tracing.utils import should_skip_tracing
+from strawberry.types.execution import ExecutionContext
+
+
+class DatadogTracingExtension(Extension):
+    def __init__(
+        self,
+        *,
+        execution_context: Optional[ExecutionContext] = None,
+    ):
+        if execution_context:
+            self.execution_context = execution_context
+
+    def on_request_start(self) -> None:
+        self._operation_name = self.execution_context.operation_name
+        span_name = (
+            f"{self._operation_name}" if self._operation_name else "Anonymous Query"
+        )
+
+        self.request_span = tracer.trace(
+            span_name,
+            resource=self.execution_context.query,
```

Comment:
@patrick91 could we make the resource the operation's name instead of the query? Would still be great to add the query as metadata but doing this would make for a better UX with Datadog's APM.

Reply:
I've done `operation-name:hash` would that work for you?

Reply:
Yes @patrick91 this works!